cdap-authorization-dataset-plugin/src/main/java/co/cask/cdap/security/authorization/ACLDataset.java

```diff
@@ -115,6 +116,28 @@ public void remove(EntityId entity) {
     }
   }
 
+  public Set<Privilege> listPrivileges(Principal principal) {
+    Set<Privilege> privileges = new HashSet<>();
+    // scan the whole table
+    Scanner scan = table.scan(null, null);
```

Comment:
I opened a jira for revoke(Principal) here https://issues.cask.co/browse/CDAP-5321